src/Interfaces.jl

```diff
@@ -83,16 +93,16 @@ function toimplements!(T, arg::Expr, shouldthrow::Bool=true)
     elseif arg.head == :(::)
         # required method definition and required return type
         nm, args = methodparts(T, arg.args[1])
-        __RT__ = arg.args[2]
-        sym = gensym()
-        return quote
-            check = $(requiredmethod(T, nm, args, shouldthrow))
-            $sym = Interfaces.returntype($nm, $args)
-            # @show $sym, $nm, $args, Interfaces.isinterfacetype($__RT__)
-            check |= Interfaces.isinterfacetype($__RT__) ?
-                Interfaces.implements($sym, $__RT__) : $sym <: $__RT__
-            check || ($shouldthrow && Interfaces.invalidreturntype($nm, $args, $sym, $__RT__))
+        annotation = arg.args[2]
+        if !isa(annotation, Symbol) && annotation.head == :where
+            # `::(T where T)` or `::(T where T<:Foo)`
+            sym = annotation.args[1]
+            __RT__ = annotation.args[2] isa Symbol ? Any : annotation.args[2].args[2]
```

Comment:
this presumes we have `T <: S` and not `T >: S`  or `S1 <: T <: S2`

...which i think is fine for now?

Reply:
Yeah, I think that's fine. probably good to throw an `@assert` in there to ensure `annotation.args[2].args[2]` exists though. otherwise the boundserror is pretty confusing